Tests/GitHubCodespaces.tests.ps1

```diff
@@ -0,0 +1,459 @@
+                RepositoryName = $defaultRepositoryName
+                Wait = $true
+            }
+            $codespace = New-GitHubCodespace @newGitHubCodespaceParms
+        }
+
+        It 'Should get no content using -Confirm:$false' {
+            # Also asserts pipeline input
+            $codespace | Remove-GitHubCodespace -Confirm:$false
+            { Get-GitHubCodespace -CodespaceName $codespace.name } | Should -Throw
+        }
+
+        It 'Should get no content using -Force' {
+            # Also assert CodespaceName input
+            Remove-GitHubCodespace -CodespaceName $codespace.name -Force
+            { Get-GitHubCodespace -CodespaceName $codespace.name } | Should -Throw
+        }
+    }
+}
+
+Describe 'GitHubCodespaces\Get-GitHubCodespace' {
+    BeforeAll {
+        # Suppress HTTP 202 warning for codespace creation
+        $WarningPreference = 'SilentlyContinue'
+
+        $newGitHubCodespaceParms = @{
+            OwnerName = $script:organizationName
+            RepositoryName = $defaultRepositoryName
+            Wait = $true
+        }
+        $null = New-GitHubCodespace @newGitHubCodespaceParms
+    }
+
+    Context 'When getting codespaces for the authenticated user' {
+        BeforeAll {
+            $codespaces = Get-GitHubCodespace |
+                Where-Object { $_.repository.name -eq $defaultRepositoryName }
+        }
+
+        It 'Should return objects of the correct type' {
+            $codespaces[0].PSObject.TypeNames[0] | Should -Be 'GitHub.Codespace'
+        }
+
+        It 'Should return one or more results' {
+            $codespaces.Count | Should -BeGreaterOrEqual 1
+        }
+
+        It 'Should return the correct properties' {
+            foreach ($codespace in $codespaces)
+            {
+                $codespace.owner.login | Should -Be $script:OwnerName
+                $codespace.repository.name | Should -Be $newGitHubCodespaceParms.RepositoryName
+            }
+        }
+    }
+
+    Context 'When getting a codespace for a specified owner and repository' {
+        BeforeAll {
+            $codespaces = Get-GitHubCodespace @newGitHubCodespaceParms
+        }
+
+        It 'Should return objects of the correct type' {
+            $codespaces[0].PSObject.TypeNames[0] | Should -Be 'GitHub.Codespace'
+        }
+
+        It 'Should return one or more results' {
+            $codespaces.Count | Should -BeGreaterOrEqual 1
+        }
+
+        It 'Should return the correct properties' {
+            foreach ($codespace in $codespaces)
+            {
+                $codespace.owner.login | Should -Be $script:OwnerName
+                $codespace.repository.name | Should -Be $newGitHubCodespaceParms.RepositoryName
+            }
+        }
+    }
+
+    Context 'When getting all codespaces for a specified organization' {
+        BeforeAll {
+            $codespaces = Get-GitHubCodespace -OrganizationName $script:organizationName
```

Comment:
This is `$null` because within the context of this test, no codespaces have been created yet.

Reply:
This test passes for me because the repository of the codespace created in the BeforeAll is in the same Organization (Settings $script:organizationName).  I am not sure how to guarantee that in both mine and your environment.  If you have an idea please share.  If not, perhaps it is best to drop this test altogether.

Reply:
I'm going to remove Context 'When getting all codespaces for a specified organization', for now.  If you insist on maintaining it, then I'll need help implementing it in a way that works in your environment.

Reply:
> This test passes for me because the repository of the codespace created in the BeforeAll is in the same Organization (Settings $script:organizationName). I am not sure how to guarantee that in both mine and your environment. If you have an idea please share. If not, perhaps it is best to drop this test altogether.

The issue is that you haven't in the context of this test, you haven't created a codespace yet.  In the root `BeforeAll`  (lines 15-35) you only created a repo.  That's the _only_ guaranteed thing that you have at the moment.  In 125-127, you're querying for codespaces within that org.  If you are expecting that to return back a non-zero number of codespaces, then it's up to you to ensure that you create a codespace within the context of that org (in that repo at 32?) before querying for the org's codespaces.